test/getGrist.ts

```diff
@@ -183,6 +192,28 @@ export class GristUtils extends GristWebDriverUtils {
     await this.waitForServer();
   }
 
+  public async sendActions(actions: UserAction[]) {
+    const result = await driver.executeAsyncScript(`
+          const done = arguments[arguments.length - 1];
+          const prom = gristDocPageModel.gristDoc.get().docModel.docData.sendActions(${JSON.stringify(actions)});
```

Comment:
This will apply changes in the front-end, but the document itself won't change - the data in sqlite and in the data engine won't change. I think for your application it will work, but it could be very confusing to someone else, and if any formulas were involved they'd go wrong. `gristDocPageModel.appModel.api.applyUserActions` would apply the actions via the server.

Reply:
It seems to be working even for formulas:
https://github.com/gristlabs/grist-widget/assets/2961294/652d3ed8-d97d-4b1f-ab4b-4e2ce9593c1d

it also clearly send data to the server via web sockets: 
![obraz](https://github.com/gristlabs/grist-widget/assets/2961294/eb4d4551-07aa-48ce-9091-df15214d629a)

Comment on code for DocData.sendActions also suggests that it send actions to the server: 
https://github.com/dsagal/grist/blob/263dea7ee587ab161332b6cf66f52fccf34eb6bf/core/app/client/models/DocData.ts#L174

Reply:
Ah, sorry, that's fine then. I confused it with a callback gristDoc has.